### PR TITLE
sg: Make detection of sg install target more robust

### PR DESCRIPTION
**Pull request by mrnugget**
With `GOBIN` and `GOPATH` unset `go install` did *not* install `sg` to
the directories described in the linked documentation.

But it did install it to the location that I could find with `go list -f
'{{.Target}}'`.

That is also the recommended way to discover the install target here:
https://golang.org/doc/tutorial/compile-install

I tested this locally in 4 different environments:

1. `asdf` Go and `GOPATH`/`GOBIN` set
2. `asdf` Go and `GOBIN` unset (`asdf` sets GOPATH by default)
3. homebrew global Go, `GOPATH`/`GOBIN` set
4. homebrew global Go, `GOPATH`/`GOBIN` unset

In all 4 cases the output by `install.sh` was correct.

### CODE REVIEW

**Review by Strum355 — APPROVED** [review]

**Review by eseliger — APPROVED** [review]

File: dev/sg/install.sh [hunk 1]
```diff
@@ -10,23 +10,21 @@ echo "Compiling..."
...
-fi
+# Let's find the install target. The documentation at
+#   https://golang.org/cmd/go/#hdr-Compile_and_install_packages_and_dependencies
+# that describes the effective $GOBIN path that's used doesn't seem correct
+# when a tool like `asdf` is used to manage Go installations.
+
+# So we use what the Go documentation recommends here
+#   https://golang.org/doc/tutorial/compile-install
+# and find the install target
+target=$(go list -f '{{.Target}}')
```
Comment (LawnGnome) [inline dev/sg/install.sh:21]: Spaces would be a problem here:

```suggestion
target="$(go list -f '{{.Target}}')"
```

File: dev/sg/install.sh [hunk 2]
```diff
@@ -10,23 +10,21 @@ echo "Compiling..."
...
 
 # Let's make sure that there's actually a binary there before we make
 # suggestions. (Unfortunately, there's no easy way to get this out of `go
 # install`, so we have to figure it out after the fact.)
-if [ ! -x "$GOBIN/sg" ]; then
-  echo "We expected to see sg in $GOBIN, but we can't find it!"
+if [ ! -x $target ]; then
```
Comment (LawnGnome) [inline dev/sg/install.sh:26]:
```suggestion
if [ ! -x "$target" ]; then
```

File: dev/sg/install.sh [hunk 3]
```diff
@@ -75,14 +73,14 @@ set -e
...
-if [ "$sg_in_path" != "$GOBIN/sg" ]; then
+if [ "$sg_in_path" != "$target" ]; then
   echo
   printf "  %s%sNOTE: this is NOT on your \$PATH.%s" "$red_bg" "$white_fg" "$reset"
   if [ -n "${sg_in_path}" ]; then
     echo "  running sg will run '$sg_in_path' instead."
   fi
   echo
-  echo "  Consider adding $GOBIN to your \$PATH for easier"
+  echo "  Consider adding $(dirname $target) to your \$PATH for easier"
```
Comment (LawnGnome) [inline dev/sg/install.sh:83]:
```suggestion
  echo "  Consider adding $(dirname "$target") to your \$PATH for easier"
```

**Review by LawnGnome — APPROVED** [review]
This definitely feels more robust. Nice one!

**Pull request merged**